SPECS/glusterfs/glusterfs.spec

```diff
@@ -74,74 +81,116 @@
+%{nil}
+%global systemd_preun() %{_sbindir}/chkconfig --del %1 >/dev/null 2>&1 || : \
+%{nil}
+%global systemd_postun_with_restart() %{_sbindir}/service %1 condrestart >/dev/null 2>&1 || : \
+%{nil}
+%global service_start()   %{_sbindir}/service %1 start >/dev/null 2>&1 || : \
+%{nil}
+%global service_stop()    %{_sbindir}/service %1 stop >/dev/null 2>&1 || : \
+%{nil}
+%global service_install() install -D -p -m 0755 %1.init %{buildroot}%2 \
+%{nil}
+# can't seem to make a generic macro that works
+%global glusterd_svcfile   %{_sysconfdir}/init.d/glusterd
+%global glusterfsd_svcfile %{_sysconfdir}/init.d/glusterfsd
+%global glustereventsd_svcfile %{_sysconfdir}/init.d/glustereventsd
+%endif
 %{!?_pkgdocdir: %global _pkgdocdir %{_docdir}/%{name}-%{version}}
 # modern rpm and current Fedora do not generate requires when the
 # provides are filtered
 %global __provides_exclude_from ^%{_libdir}/glusterfs/%{version}/.*$
+%global bashcompdir %(pkg-config --variable=completionsdir bash-completion 2>/dev/null)
+%if "%{bashcompdir}" == ""
+%global bashcompdir ${sysconfdir}/bash_completion.d
+%endif
+# Skip regression-tests because they previously were excluded in
+# CBL-Mariner glusterfs and the new version introduces a dependency
+# (dbench)
+%global _without_regression_tests true
+# Explicitly require rpcgen
+%global _require_rpcgen true
 ##-----------------------------------------------------------------------------
 ## All package definitions should be placed here and keep them sorted
 ##
 Summary:        Distributed File System
 Name:           glusterfs
-Version:        7.9
-Release:        5%{?dist}
+Version:        11.1
+Release:        1%{?dist}
 License:        GPLv2 OR LGPLv3+
 Vendor:         Microsoft Corporation
 Distribution:   Mariner
 URL:            https://docs.gluster.org/
-Source0:        https://download.gluster.org/pub/gluster/%{name}/7/%{version}/%{name}-%{version}.tar.gz
+Source0:        https://download.gluster.org/pub/gluster/%{name}/11/%{version}/%{name}-%{version}.tar.gz
 Source1:        glusterd.sysconfig
 Source2:        glusterfsd.sysconfig
 Source7:        glusterfsd.service
 Source8:        glusterfsd.init
-Patch0001:      0001-rpc-rpc-lib-src-Makefile.am.patch
+
+# Patch created for Mariner to fix build break related to
+# including eventtypes.h
+Patch001:       include-eventtypes-always.patch
+
+BuildRoot:        %(mktemp -ud %{_tmppath}/%{name}-%{version}-%{release}-XXXXXX)
 
 BuildRequires:  bison
 BuildRequires:  flex
 BuildRequires:  gcc
-BuildRequires:  libacl-devel
 BuildRequires:  libaio-devel
-BuildRequires:  libtool
+BuildRequires:  libacl-devel
 BuildRequires:  libuuid-devel
+BuildRequires:  libtool
 BuildRequires:  libxml2-devel
 BuildRequires:  make
 BuildRequires:  ncurses-devel
 BuildRequires:  openssl-devel
+BuildRequires:  openssl
 BuildRequires:  python%{_pythonver}-devel
 BuildRequires:  readline-devel
-BuildRequires:  rpcgen
-BuildRequires:  systemd
 BuildRequires:  userspace-rcu-devel >= 0.7
```

Comment:
Nit-pick: can we keep the dependencies in alphabetical order as much as possible?